pkg/http/server.go

```diff
@@ -69,6 +107,16 @@ func (s *server) setupListener(opts Opts) error {
 		return nil
 	}
 
+	if strings.HasPrefix(strings.ToLower(opts.BindAddress), "fd:") {
```

Comment:
Add a comment before this, explaining that the systemd socket activation needs to be handled separately. Want to make it clear in the future why we are branching here and what the use case is, I'm unlikely to remember why `fd:` is a special case

Reply:
Added